llvm/lib/Target/X86/X86AsmPrinter.cpp

```diff
@@ -97,7 +97,17 @@ void processInstructions(
       const MachineOperand Rhs = Instr.getOperand(1);
       assert(Lhs.isReg() && "Is register.");
       assert(Rhs.isReg() && "Is register.");
-      SpillMap[Lhs.getReg()] = Rhs.getReg();
+      if (SpillMap.count(Rhs.getReg()) > 0) {
+        // If the RHS has a mapping, apply the same mapping to the new register.
```

Comment:
Does this mean we also need to track the case where a later assignment to this register "unmaps" the previous mapping?

Reply:
Yes, this is already done in https://github.com/ykjit/ykllvm/pull/58/files/2571222a1236501bf49f79f6b55d2aa5dc71fc7a#diff-f9637e37f0c0135987f07d887a8b95370815ea80913c2b141432fee0d20abbefR113. Whenever overwrite a register, we find all mappings that map to it, and remove them.